lib/src/rules/avoid_types_as_parameter_names.dart

```diff
@@ -75,11 +84,30 @@ class _Visitor extends SimpleAstVisitor<void> {
   }
 
   bool _isTypeName(AstNode scope, SimpleIdentifier node) {
+    var parent = scope.parent;
+    if (parent is FunctionExpression) {
```

Comment:
I don't believe you need either lines 87-99 or lines 107-112 either. I believe that the easier fix is to extend the test on line 103 to include `element is TypeParameterElement`.

Reply:
I am afraid `context.resolveNameInScope` always return `none` state with the new scenarios.

Reply:
Oh. That sounds like a bug in `resolveNameInScope`. The names of type parameters are definitely in scope in the formal parameter list. I've created an issue to track it (https://github.com/dart-lang/sdk/issues/46753).

Given that, please add a TODO to remove this code when the bug has been fixed (and referencing that issue).